src/main/java/org/benetech/servicenet/service/ClientService.java

```diff
@@ -45,10 +47,14 @@
      * @param clientDto client to create
      * @return created client
      */
-    public ClientDTO createClient(ClientDTO clientDto) {
-        ClientDTO result = authClient.createClient(clientDto);
-        this.createOrUpdateClientProfile(result, clientDto);
-        return this.getRealDto(result);
+    public ClientDTO createClient(ClientDTO clientDto) throws BadRequestAlertException {
+        try {
+            ClientDTO result = authClient.createClient(clientDto);
+            this.createOrUpdateClientProfile(result, clientDto);
+            return this.getRealDto(result);
+        } catch (Exception e) {
+            throw new BadRequestAlertException(ErrorConstants.LOGIN_ALREADY_USED_TYPE, "Client id already used!", "clientManagement", "clientexists");
```

Comment:
I think we should check for response status, we don't want to throw 'client id already used' for any error
it can be done by creating a custom error decoder, I believe spring picks it up if you make it a @Bean
https://github.com/OpenFeign/feign/wiki/Custom-error-handling